### PR TITLE
Fix #2333 CPCSS to retain > child selector in CSS.

### LINKED ISSUE #2333: Character ">" in the CPCSS Fallback Field is Replaced by its HTML Name "&gt;"
**Describe the bug**
The `>` symbol is being replaced by its HTML name, `&gt;` when `>` is used in the **Fallback critical CSS** field. 

**To Reproduce**
1. Use the Fallback critical CSS feature (disable automatic CPCSS generation, delete content in wp-content/cache/critical-css/1 folder, add the custom code in the field).
2. Add any CSS rule with the `<` character:
image
3. Save changes in WP Rocket.
4. See the error in the source code:
image

**Expected behavior**
The  `<` character should not be replaced.

**Additional context**
Happens in the 3.4.4 version (latest) and has been tested in customer's site, my test site and the Mega/multisite site.
**Slack thread:** https://wp-media.slack.com/archives/C43T1AYMQ/p1582209375163400.
**Ticket:** https://secure.helpscout.net/conversation/1088000268/146959?folderId=2675957.

**Backlog Grooming**
- [x] Reproduce the problem
- [x] Identify the root cause
- [x] Scope a solution
- [x] Estimate the effort

### PR DESCRIPTION
**Pull request by crystinutzaa**
Closes #2333 

**Reproduce the issue** ✅ 
Reproduced on my local test site

**Identify the root cause** ✅ 
This is the line: https://github.com/wp-media/wp-rocket/blob/8cf329ca6f1e7ad9145c80499cfd4be98e702e3b/inc/classes/admin/settings/class-settings.php#L315

The use of `wp_kses()` is changin the character to it's HTML-encoded version on save to the database. This is causing the output to be incorrect once inserted in the HTML.

**Scope a solution** ✅ 
Like we did for the output issue in #1654, we can replace `wp_kses()` by `wp_strip_all_tags()` before saving.

**Estimate the effort** ✅ 

`Effort: [S]`

The code change itself is very low effort. The current state of the `sanitize_callback()` method though might make writing automatic tests for this change very complicated.

## TODO

- [x] Write unit tests
- [x] Write integration tests
- [x] Add admin integration test case
- [ ] Do QA

### CODE REVIEW

**Comment by hellofromtonya** [comment]
@crystinutzaa The new `AdminTestCase` from wp-media/phpunit is now applied to the integration tests. I also did a little simplification in the test data too. Let me know if you need anything else.

**Comment by hellofromtonya** [comment]
This PR is ready for QA. @arunbasillal @piotrbak or @vmanthos -- one of you please self-assign and take care of this one if possible by end of day Friday. Thanks!

File: licence-data.php [hunk 1]
```diff
@@ -1,18 +0,0 @@
-<?php
```
Comment (Tabrisrp) [inline licence-data.php]: this file should not be deleted

File: tests/Integration/inc/classes/admin/settings/Settings/sanitizeCallback.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ * @covers \WP_Rocket\Admin\Settings::sanitize_callback
+ * @group  AdminOnly
+ * @group  Settings
+ */
+class Test_SanitizeCallback extends AdminTestCase {
+
+	public function setUp() {
+		parent::setUp();
+
+		remove_action( 'admin_init', 'send_frame_options_header' );
```
Comment (Tabrisrp) [inline tests/Integration/inc/classes/admin/settings/Settings/sanitizeCallback.php]: is it still needed?

File: tests/Integration/inc/classes/admin/settings/Settings/sanitizeCallback.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	}
+
+	public function tearDown() {
+		parent::tearDown();
+
+		add_action( 'admin_init', 'send_frame_options_header' );
+	}
```
Comment (Tabrisrp) [inline tests/Integration/inc/classes/admin/settings/Settings/sanitizeCallback.php]: same here

File: tests/Unit/inc/classes/admin/settings/Settings/sanitizeCallback.php [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+			return trim( $string );
+		} );
+
+		Functions\when( 'sanitize_email' )->alias( function( $email ) {
+			return $email;
+		} );
+
+		Functions\when( 'sanitize_text_field' )->alias( function( $text ) {
+			return $text;
+		} );
```
Comment (Tabrisrp) [inline tests/Unit/inc/classes/admin/settings/Settings/sanitizeCallback.php]: For those you can use the `returnArg()` method instead

File: tests/Unit/inc/classes/admin/settings/Settings/sanitizeCallback.php [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+		$options = $this->createMock( 'WP_Rocket\Admin\Options_Data' );
+		$map     = [
+			[ 'defer_all_js', 0, 1 ],
+			[ 'defer_all_js_safe', 0, 1 ],
+			[ 'purge_cron_interval', 0, 0 ],
+			[ 'purge_cron_unit', 0, 'HOUR_IN_SECONDS' ],
+			[ 'schedule_automatic_cleanup', 0, 0 ],
+			[ 'automatic_cleanup_frequency', 'daily', 'weekly' ],
+		];
+		$options->method( 'get' )->will( $this->returnValueMap( $map ) );
```
Comment (Tabrisrp) [inline tests/Unit/inc/classes/admin/settings/Settings/sanitizeCallback.php]: I don't think you need that for this test

File: tests/Unit/inc/classes/admin/settings/Settings/sanitizeCallback.php [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+			if ( $remove_breaks ) {
+				$string = preg_replace( '/[\r\n\t ]+/', ' ', $string );
+			}
+
+			return trim( $string );
+		} );
+
+		Functions\when( 'sanitize_email' )->alias( function( $email ) {
+			return $email;
+		} );
```
Comment (Tabrisrp) [inline tests/Unit/inc/classes/admin/settings/Settings/sanitizeCallback.php]:
```suggestion
		Functions\when( 'sanitize_email' )->returnArg();
```

**Comment by hellofromtonya** [comment]
This PR is ready for QA. @arunbasillal @piotrbak or @vmanthos -- one of you please self-assign. Thanks!

**Comment by arunbasillal** [comment]
Starting QA on this one

**Comment by arunbasillal** [comment]
- Reproduced the issue with the current `master` - https://jmp.sh/lSYLLGu
- Switching to this branch had the expected result - https://jmp.sh/LK6e5uG

All good 👍

**Pull request merged**